dig.go

```diff
@@ -211,10 +211,51 @@ func New(opts ...Option) *Container {
 	return c
 }
 
-// A VisualizeOption modifies the default behavior of Visualize. It's included
-// for future functionality; currently, there are no concrete implementations.
+// A VisualizeOption modifies the default behavior of Visualize.
 type VisualizeOption interface {
-	unimplemented()
+	applyVisualizeOption(*visualizeOptions)
+}
+
+type visualizeOptions struct {
+	VisualizeError error
+}
+
+type visualizeOptionFunc func(*visualizeOptions)
+
+func (f visualizeOptionFunc) applyVisualizeOption(opts *visualizeOptions) { f(opts) }
+
+// VisualizeError is a VisualizeOption that specifies the error updates to the graph
+// that should be made before visualizing it.
+func VisualizeError(err error) VisualizeOption {
+	return visualizeOptionFunc(func(opts *visualizeOptions) {
+		opts.VisualizeError = err
+	})
+}
+
+func updateGraph(dg *dot.Graph, err error) error {
+	var errors []errVisualizer
+	// Unwrapping the error until we reach the root cause.
+	for {
+		if ev, ok := err.(errVisualizer); ok {
+			errors = append(errors, ev)
+		}
+		e, ok := err.(causer)
+		if !ok {
+			break
+		}
+		err = e.cause()
+	}
+
+	if len(errors) == 0 {
+		return fmt.Errorf("cannot visualize error: %v", err)
```

Comment:
Since the modification for provide errors still hasn't been implemented, I agree that we can skip the irrelevant errors and just output the unchanged graph.